fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/api/client/gametest/v1/ClientGameTestInput.java

```diff
@@ -0,0 +1,318 @@
+	 *
+	 * @param keyBindingGetter The function to get the key binding from the game options
+	 * @see #holdKey(Function)
+	 * @see #pressKey(KeyBinding)
+	 */
+	void pressKey(Function<GameOptions, KeyBinding> keyBindingGetter);
+
+	/**
+	 * Presses and releases a key or mouse button.
+	 *
+	 * @param key The key or mouse button to press.
+	 * @see #holdKey(InputUtil.Key)
+	 */
+	void pressKey(InputUtil.Key key);
+
+	/**
+	 * Presses and releases a key.
+	 *
+	 * @param keyCode The key code of the key to press
+	 * @see #holdKey(int)
+	 */
+	void pressKey(int keyCode);
+
+	/**
+	 * Presses and releases a mouse button.
+	 *
+	 * @param button The mouse button to press
+	 * @see #holdMouse(int)
+	 */
+	void pressMouse(int button);
+
+	/**
+	 * Holds a key binding for the specified number of ticks and then releases it. Waits until this process is finished.
+	 * The key binding must be bound.
+	 *
+	 * @param keyBinding The key binding to hold
+	 * @param ticks The number of ticks to hold the key binding for
+	 * @see #holdKey(KeyBinding)
+	 * @see #holdKeyFor(Function, int)
+	 */
+	void holdKeyFor(KeyBinding keyBinding, int ticks);
+
+	/**
+	 * Holds a key binding for the specified number of ticks and then releases it. Waits until this process is finished.
+	 * The key binding must be bound.
+	 *
+	 * @param keyBindingGetter The key binding to hold
+	 * @param ticks The number of ticks to hold the key binding for
+	 * @see #holdKey(Function)
+	 * @see #holdKeyFor(Function, int)
+	 */
+	void holdKeyFor(Function<GameOptions, KeyBinding> keyBindingGetter, int ticks);
+
+	/**
+	 * Holds a key or mouse button for the specified number of ticks and then releases it. Waits until this process is
+	 * finished.
+	 *
+	 * @param key The key or mouse button to hold
+	 * @param ticks The number of ticks to hold the key or mouse button for
+	 * @see #holdKey(InputUtil.Key)
+	 */
+	void holdKeyFor(InputUtil.Key key, int ticks);
+
+	/**
+	 * Holds a key for the specified number of ticks and then releases it. Waits until this process is finished.
+	 *
+	 * @param keyCode The key code of the key to hold
+	 * @param ticks The number of ticks to hold the key for
+	 * @see #holdKey(int)
+	 */
+	void holdKeyFor(int keyCode, int ticks);
+
+	/**
+	 * Holds a mouse button for the specified number of ticks and then releases it. Waits until this process is
+	 * finished.
+	 *
+	 * @param button The mouse button to hold
+	 * @param ticks The number of ticks to hold the mouse button for
+	 * @see #holdMouse(int)
+	 */
+	void holdMouseFor(int button, int ticks);
+
+	/**
+	 * Types a code point (character). Useful for typing in text boxes.
+	 *
+	 * @param codePoint The code point to type
+	 * @see #typeChars(String)
+	 */
+	void typeChar(int codePoint);
+
+	/**
+	 * Types a sequence of code points (characters) one after the other. Useful for typing in text boxes.
+	 *
+	 * @param chars The code points to type
+	 */
+	void typeChars(String chars);
```

Comment:
It should do. As the documentation states, it works with code points, not 16-bit `char`s. The implementation uses `String.chars()` which is a stream of the string's code points, rather than `char`s.

Reply:
Yup, works fine.

![chat_message_sent](https://github.com/user-attachments/assets/2484b1e1-4269-4d3f-bf24-aa6747cb490b)